### PR TITLE
cleanup(tgc): retry fetch full resource API in convert.go

### PR DESCRIPTION
**Pull request by iyabchen**
b/282940502 retry fetch full resource API in convert.go

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/tgc/convert.go [hunk 1]
```diff
@@ -172,13 +173,20 @@ func (c *Converter) addDelete(rc *tfjson.ResourceChange) error {
...
-
 			key := converted.Type + converted.Name
 			var existingConverterAsset *resources.Asset
 			if existing, exists := c.assets[key]; exists {
 				existingConverterAsset = &existing.converterAsset
 			} else if !c.offline {
-				asset, err := converter.FetchFullResource(rd, c.cfg)
+				var asset cai.Asset
+				var err error
+				err = transport_tpg.Retry(transport_tpg.RetryOptions{
```
Comment (melinath) [inline mmv1/third_party/tgc/convert.go:183]: Rather than having the retry handling here, it should be inside the converter.FetchFullResource function so that the error retry / abort predicates can be included - similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/a8ab34eb06ae386e8394ba6a690c33d84a235aa6/mmv1/templates/terraform/resource.erb#L271 (except that calls SendRequestOptions, which is a wrapper around Retry. We can still use Retry directly, we just need to make sure we can supply it with the retry / abort predicates, which are per-resource, so are not knowable here.)

File: mmv1/third_party/tgc/convert.go [hunk 2]
```diff
@@ -190,14 +198,14 @@ func (c *Converter) addDelete(rc *tfjson.ResourceChange) error {
...
 				}
-				if existingConverterAsset != nil {
-					converted = converter.MergeDelete(*existingConverterAsset, converted)
-					augmented, err := c.augmentAsset(rd, c.cfg, converted)
-					if err != nil {
-						return err
-					}
-					c.assets[key] = augmented
+			}
+			if existingConverterAsset != nil {
```
Comment (melinath) [inline mmv1/third_party/tgc/convert.go:202]: This does seem like a bug but it might be nice to fix in a separate PR for clarity.

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/third_party/tgc/convert.go (hunk 1 above)
Comment (iyabchen) [reply mmv1/third_party/tgc/convert.go:183]: https://github.com/hashicorp/terraform-provider-google-beta/blob/main/google-beta/transport/transport.go#L55
I think this PR can be abandoned. 
As long as using transport_tpg.SendRequest, it has retry logic baked in. 
I checked all the FetchFullResource calls, and they are all eventually using SendRequest.

File: mmv1/third_party/tgc/convert.go (hunk 2 above)
Comment (iyabchen) [reply mmv1/third_party/tgc/convert.go:202]: https://github.com/GoogleCloudPlatform/magic-modules/pull/9790

File: mmv1/third_party/tgc/convert.go (hunk 1 above)
Comment (melinath) [reply mmv1/third_party/tgc/convert.go:183]: Makes sense! SGTM.
